spec/controllers/state_file/questions/unemployment_controller_spec.rb

```diff
@@ -23,7 +28,33 @@
     end
   end
 
+  describe "#new" do
+    context "before_action redirect_if_df_data_required" do
+      before do
+        intake.update(df_data_import_succeeded_at: nil)
+      end
+
+      it "redirects to login on edit" do
+        get :new
+
+        expect(response).to redirect_to StateFile::StateFilePagesController.to_path_helper(action: :login_options)
+      end
+    end
+  end
+
```

Comment:
[q] is the reason we don't include `get :new` or `get :index` into the shared example is that some of these controllers just do not have these actions? 

this might be a bit going overboard but when if we just pass in the action to the `it_behaves_like` as an argument?

`it_behaves_like :df_data_required, true, :az, :action, :record_type`? Not super passionate about it but just a suggestion to make it easier to add to future controllers (altho idk if we'll ever add more)

Reply:
answer to the initial question is yes, most controllers don't have anything beyond edit and update.

i thought briefly about passing in the action but we'd then have to have at least 2 `it_behaves_like` lines per controller spec (also the shared example would have to make some distinction between `get` and `post` which is a bit more complicated). i'm not sure which approach i prefer tbh so ended up leaving it as is.

Reply:
np! ty for listing out the possibilities. I'm good with current approach, especially as I don't think we'll be adding more controllers (at least at this time)